rc/core/sh-lint.kak

```diff
@@ -0,0 +1,88 @@
+# ex: '-e SC2087'
+decl str sh_options
+
+decl -hidden line-flags sh_flags
+decl -hidden str sh_errors
+
+def sh-parse -params 0..1 -docstring 'Parse the contents of the current buffer with shellcheck' %{
+    %sh{
+        dir=$(mktemp -d -t kak-sh.XXXXXXXX)
+        mkfifo "$dir"/fifo
+        printf '%s\n' "eval -no-hooks write $dir/buf"
+
+        printf '%s\n' "eval -draft %{
+                  edit! -fifo $dir/fifo *sh-output*
+                  set buffer filetype make
+                  set buffer _make_current_error_line 0
+                  hook -group fifo buffer BufCloseFifo .* %{
+                      nop %sh{ rm -r '$dir' }
+                      rmhooks buffer fifo
+                  }
+              }"
+
+        { # do the parsing in the background and when ready send to the session
+
+        shellcheck -fgcc ${kak_opt_sh_options} "$dir"/buf > "$dir"/stderr
+        printf '%s\n' "eval -client $kak_client echo 'shellcheck parsing done'" | kak -p "$kak_session"
+
+        # Update the gutter with errors and warnings: line3|{red}:line11|{yellow}
+        flags=$(awk -F: '
+                    /:[0-9]+:([0-9]+:)? (fatal )?error/ { print $2"|{red}█" }
+                    /:[0-9]+:([0-9]+:)? (warning|note)/ { print $2"|{yellow}█" }
+                ' "$dir"/stderr | paste -s -d ':')
+
+        # 1,err1
+        # 2,err2
+        # ...
+        errors=$(awk -F: '
+                    /:[0-9]+:([0-9]+:)? ((fatal )?error|warning|note)/ { print $2","substr($4,2)":"$5 }
+                 ' "$dir"/stderr | sort -n)
```

Comment:
similarly, maybe we can join these two awk scripts to build two different variables, (and maybe even the two option set commands) in one go. (Not sure its worth it, maybe it will clean up the code, maybe not, so its really up to you).